### ISSUE TITLE
Sidenotes do not avoid each other when called inside definition

### ISSUE THREAD

**Issue by dwarfmaster** [fmarotta/kaobook#282]
### Description
When adding two sidenotes too close to each other, kaobook manages to place them so that they avoid each other. However, if they are placed inside a definition, it seems the floating behaviour is disabled.

### Minimal Working Example

```latex
\documentclass{kaobook}
\usepackage[english]{babel}
\usepackage{blindtext}
\usepackage{kaobiblio}
\usepackage[framed=true]{kaotheorems}
\usepackage{kaorefs}

\begin{document}
\begin{definition}
Some text\sidenote{Hello}\sidenote{World}
\end{definition}
\end{document}
```

**Expected behavior:**
I would expect `Hello` and `World` not to overlap.

**Actual behavior:**
`Hello` and `World` actually overlap.

**Comment by dwarfmaster** [comment]
Turns out it is linked to the `mdframed` environment, it works as expected with `framed=false`.

**Comment by fmarotta** [comment]
Yep, normally the sidenotes avoid each other because they are so-called "floating". However, we turn the floating off when they are inside other floating environments (such as figures and tables) or `tcolorbox` environments. There is some context in #39, however I think that issue dates back to the `mdframed` era, that is, before we moved to `tcolorbox`. We should indeed check whether the "floats lost" issue of #39 still occurs with `tcolorbox`..
